packages/pages-shared/asset-server/handler.ts

```diff
@@ -364,13 +365,14 @@ export async function generateHandler<
 
 		if (
 			earlyHintsCache &&
-			isHTMLContentType(response.headers.get("Content-Type"))
+			isHTMLContentType(response.headers.get("Content-Type")) &&
+			eTag
 		) {
 			const preEarlyHintsHeaders = new Headers(headers);
 
 			// "Early Hints cache entries are keyed by request URI and ignore query strings."
 			// https://developers.cloudflare.com/cache/about/early-hints/
-			const earlyHintsCacheKey = `${protocol}//${host}${pathname}`;
+			const earlyHintsCacheKey = `${protocol}//${host}/${eTag}`;
```

Comment:
I could do [something like we did for asset preservation v2](https://github.com/cloudflare/workers-sdk/pull/7544/files) where we simultaneously checked the old and new formats to prevent a flood of new cache misses as we roll this out.